### PR TITLE
Features/delly reformat recipe

### PR DESCRIPTION
**Pull request by northwestwitch**
I want to kill myself, the bug was the position of a parenthesis! :-/

### CODE REVIEW

**Comment by henrikstranneheim** [comment]
At least you tracked it down 😄

**Comment by northwestwitch** [comment]
Yes, let's be positive!

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 1]
```diff
@@ -0,0 +1,1069 @@
...
+            default     => 0,
+            allow       => qr/ ^\d+$ /xsm,
+            strict_type => 1,
+            store       => \$xargs_file_counter,
+        },
+    };
+
+    check( $tmpl, $arg_href, 1 ) or croak q{Could not parse arguments!};
+
+    ### SET IMPORT IN ALPHABETIC ORDER
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: Remove

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 2]
```diff
@@ -0,0 +1,1069 @@
...
+    use MIP::Delete::List qw{ delete_contig_elements };
+    use MIP::Get::File qw{ get_file_suffix get_merged_infile_prefix };
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: Add the new sub get_module_parameter. See code template

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 3]
```diff
@@ -0,0 +1,1069 @@
...
+    ## Retrieve logger object
+    my $log = Log::Log4perl->get_logger(q{MIP});
+
+    ## Set MIP program name
+    my $mip_program_name = q{p} . $program_name;
+    my $mip_program_mode = $active_parameter_href->{$mip_program_name};
+
+    ## Unpack parameters
+    my $job_id_chain = $parameter_href->{$mip_program_name}{chain};
+    my $core_number =
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: Use get_module_parameter here

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 4]
```diff
@@ -0,0 +1,1069 @@
...
+    my $FILEHANDLE      = IO::Handle->new();
+    my $XARGSFILEHANDLE = IO::Handle->new();
+
+    my $program_directory =
+      catfile( $outaligner_dir, $program_outdirectory_name );
+
+    ## Creates program directories (info & programData & programScript), program script filenames and writes sbatch header
+    my ( $file_path, $program_info_path ) = setup_script(
+        {
+            active_parameter_href => $active_parameter_href,
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: Order alphabetically

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 5]
```diff
@@ -0,0 +1,1069 @@
...
+            job_id_href           => $job_id_href,
+            FILEHANDLE            => $FILEHANDLE,
+            directory_id          => $family_id,
+            program_name          => $program_name,
+            program_directory     => $program_directory,
+            core_number           => $core_number,
+            process_time          => $time,
+            temp_directory        => $temp_directory,
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: Add source env command. See code recipe template

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 6]
```diff
@@ -0,0 +1,1069 @@
...
+    # Used downstream
+    $parameter_href->{$mip_program_name}{indirectory} = $outfamily_directory;
+
+    ## Tags
+    my $outfile_tag =
+      $file_info_href->{$family_id}{$mip_program_name}{file_tag};
+
+    ## Files
+    my $outfile_prefix = $family_id . $outfile_tag . $UNDERSCORE . $call_type;
+    my $outfile_path_prefix = catfile( $temp_directory, $outfile_prefix );
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: '##' Paths

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 7]
```diff
@@ -0,0 +1,1069 @@
...
+    my $file_suffix = $parameter_href->{$mip_program_name}{outfile_suffix};
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: Move to under '## Assign suffix'

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm (hunk 4 above)
Comment (henrikstranneheim) [reply lib/MIP/Recipes/Analysis/Delly_reformat.pm]: throughout rest of script

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 8]
```diff
@@ -0,0 +1,1069 @@
...
+            parameter_href => $parameter_href,
+            suffix_key     => q{variant_file_suffix},
+            job_id_chain   => $job_id_chain,
+            file_suffix    => $file_suffix,
+        }
+    );
+
+    ### Update contigs
+    ## Removes an element from array and return new array while leaving orginal elements_ref untouched
+# Skip contig Y along with MT throughout since sometimes there are no variants particularly for INS
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: Indent

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 9]
```diff
@@ -0,0 +1,1069 @@
...
+        }
+    );
+
+    ## Collect files and suffix for all sample_ids
+    my %infile_path_prefix;
+    my %file_path_prefix;
+    my %suffix;
+    my @program_tag_keys = (qw{ pgatk_baserecalibration pdelly_call });
+
+    foreach my $sample_id ( @{ $active_parameter_href->{sample_ids} } ) {
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: SAMPLE_ID:

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 10]
```diff
@@ -0,0 +1,1069 @@
...
+        my $insample_directory_bam =
+          catdir( $active_parameter_href->{outdata_dir},
+            $sample_id, $outaligner_dir );
+
+        my $insample_directory_bcf =
+          catdir( $active_parameter_href->{outdata_dir},
+            $sample_id, $outaligner_dir,
+            $parameter_href->{pdelly_call}{outdir_name} );
+
+        foreach my $infile_tag_key (@program_tag_keys) {
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: INFILE_TAG:

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 11]
```diff
@@ -0,0 +1,1069 @@
...
+                                FILEHANDLE  => $FILEHANDLE,
+                                infile_path => $infile_path,
+                                outfile_path =>
+                                  $active_parameter_href->{temp_directory},
+                            }
+                        );
+                    }
+                }
+            }
+            else {    #BAMs
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: Move comment to below else

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 12]
```diff
@@ -0,0 +1,1069 @@
...
+                            stderrfile_path => $xargs_file_path_prefix
+                              . $DOT
+                              . $contig
+                              . $DOT
+                              . $sv_type
+                              . $DOT
+                              . q{stderr.txt},
+                            sv_type    => $sv_type,
+                            min_size   => 0,
+                            max_size   => 100000000,
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: Use underscore to seperate numbers

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 13]
```diff
@@ -0,0 +1,1069 @@
...
+        migrate_file(
+            {
+                infile_path  => $outfile_path_prefix . $outfile_suffix,
+                outfile_path => $outfamily_directory,
+                FILEHANDLE   => $FILEHANDLE,
+            }
+        );
+        say {$FILEHANDLE} q{wait}, $NEWLINE;
+
+        if ( $active_parameter_href->{$mip_program_name} == 1 ) {
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: $active_parameter_href->{$mip_program_name} -> $mip_program_mode

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 14]
```diff
@@ -0,0 +1,1069 @@
...
+
+            add_program_outfile_to_sample_info(
+                {
+                    sample_info_href => $sample_info_href,
+                    program_name     => q{delly},
+                    outdirectory     => $outfamily_directory,
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: Use path instead of file and dir

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 15]
```diff
@@ -0,0 +1,1069 @@
...
+                    log              => $log,
+                    sbatch_file_name => $file_path,
+                }
+            );
+        }
+
+        close $FILEHANDLE or $log->logcroak(q{Could not close FILEHANDLE});
+        close $XARGSFILEHANDLE
+          or $log->logcroak(q{Could not close $XARGSFILEHANDLE});
+
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: Remove empty line

File: lib/MIP/Recipes/Analysis/Delly_reformat.pm [hunk 16]
```diff
@@ -0,0 +1,1069 @@
...
+    }
+
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Delly_reformat.pm]: Remove empty line

**Comment by henrikstranneheim** [comment]
👍

**Pull request merged**